icepyx/quest/dataset_scripts/argo.py

```diff
@@ -0,0 +1,296 @@
+from icepyx.quest.dataset_scripts.dataset import DataSet
+from icepyx.core.spatial import geodataframe
+import requests
+import pandas as pd
+import os
+import numpy as np
+
+
+class Argo(DataSet):
+    """
+    Initialises an Argo Dataset object
+    Used to query physical Argo profiles
+    -> biogeochemical Argo (BGC) not included
+
+    Examples
+    --------
+    # example with profiles available
+    >>> reg_a = Argo([-154, 30,-143, 37], ['2022-04-12', '2022-04-26'])
+    >>> reg_a.search_data()
+    >>> print(reg_a.profiles[['pres', 'temp', 'lat', 'lon']].head())
+        pres    temp     lat      lon
+    0   3.9  18.608  33.401 -153.913
+    1   5.7  18.598  33.401 -153.913
+    2   7.7  18.588  33.401 -153.913
+    3   9.7  18.462  33.401 -153.913
+    4  11.7  18.378  33.401 -153.913
+
+    # example with no profiles
+    >>> reg_a = Argo([-55, 68, -48, 71], ['2019-02-20', '2019-02-28'])
+    >>> reg_a.search_data()
+    Warning: Query returned no profiles
+    Please try different search parameters
+
+
+    See Also
+    --------
+    DataSet
+    GenQuery
+    """
+
+    # DevNote: it looks like ArgoVis now accepts polygons, not just bounding boxes
+    def __init__(self, boundingbox, timeframe):
+        super().__init__(boundingbox, timeframe)
+        assert self._spatial._ext_type == "bounding_box"
+        self.profiles = None
+
+    def search_data(self, params=["all"], presRange=None, printURL=False) -> str:
```

Comment:
@kelseybisson @RomiP Any idea what the `presRange=None` is here? Was it a way to search by depth? What type of input did it accept (e.g. list of floats)? I can't find any mention of it, so I'm wondering if I can remove it.

Reply:
Yes, it's to specify if you want a subset of profiles (i.e. upper 200m only). We can probably safely remove it but there may be an argovis requirement to specify a pressure range, in which case 'None' is probably needed to stay there. I'm not sure what the default range is...

Reply:
Thanks - I'm guessing the default is to not filter/subset by pressure (so give you everything)... I was able to find mention of it elsewhere (there are currently no docs for the api - the url leads to a blank page - so I have to use the examples or helper functions themselves to see what the accepted keywords and formats are). But I just used it in a query and got back just the requested pressures.